File: src/org/usfirst/frc5293/input/util/UpDownInput.java
package org.usfirst.frc5293.input.util;

import edu.wpi.first.wpilibj.Joystick;
import edu.wpi.first.wpilibj.buttons.JoystickButton;

public abstract class UpDownInput {

    private static JoystickButton upButton;
    private static JoystickButton downButton;

    public UpDownInput(Joystick joystick, int upButtonNumber, int downButtonNumber) {
        upButton = new JoystickButton(joystick, upButtonNumber);
        downButton = new JoystickButton(joystick, downButtonNumber);
    }

    public JoystickButton getUpButton() {
        return upButton;
    }

    public JoystickButton getDownButton() {
        return downButton;
    }
}

File: src/main/kotlin/org/usfirst/frc5293/input/util/UpDownInput.kt
package org.usfirst.frc5293.input.util

import edu.wpi.first.wpilibj.Joystick
import edu.wpi.first.wpilibj.buttons.JoystickButton

abstract class UpDownInput(joystick: Joystick, upButtonNumber: Int, downButtonNumber: Int) {

    val upButton = JoystickButton(joystick, upButtonNumber)
    val downButton = JoystickButton(joystick, downButtonNumber)
}
